app/renderer/components/lobby/modals/SessionSettings.tsx

```diff
@@ -153,21 +155,37 @@ class SessionSettings extends Component<PrivateProps, IState> {
   }
 
   private renderUserOpts() {
-    const userOpts = []
+    if (this.props.settings.sessionMode === SessionMode.Offline) {
+      return false
+    }
 
-    for (let i = 2; i <= USERS_MAX; i = i << 1) {
-      userOpts.push(i)
+    const dispatch = this.props.dispatch!
+    const userOpts: JSX.Element[] = []
+
+    const updateMaxUsers = (ev: ChangeEvent<{ children: React.ReactNode }>) => {
+      const newValue = (ev.currentTarget as HTMLSelectElement).value
+      dispatch(setSessionData({ maxUsers: parseInt(newValue) }))
     }
 
-    const elems = userOpts.map(numUsers => {
-      return <option value={numUsers}>{numUsers} users</option>
-    })
+    const addOption = (opt: number, customLbl: string | null) => {
+      const element = <option key={opt} value={opt} selected={opt === this.props.maxUsers}>
+        {customLbl || `${opt} ${t('users')}`}
+      </option>
+
+      userOpts.push(element)
+    }
+
+    for (let i = 2; i <= USERS_MAX; i = i << 1) {
+      addOption(i, null)
+    }
+    addOption(Infinity, t('unlimitedUsers'))
```

Comment:
`Infinity` isn't JSON-serializable, so we'll need another value to represent no max-users. Something like `const MAX_USERS_INFINITE = -1` would work. This way the setting will be persisted when restarting Metastream.